span-processing-utils/src/main/java/org/hypertrace/config/span/processing/utils/SpanFilterMatcher.java

```diff
@@ -99,33 +99,34 @@ public boolean matches(String lhs, SpanFilterValue rhs, RelationalOperator relat
   }
 
   private boolean matches(String lhs, String rhs, RelationalOperator relationalOperator) {
-    switch (relationalOperator) {
-      case RELATIONAL_OPERATOR_CONTAINS:
-        return lhs.contains(rhs);
-      case RELATIONAL_OPERATOR_EQUALS:
-        return lhs.equals(rhs);
-      case RELATIONAL_OPERATOR_NOT_EQUALS:
-        return !lhs.equals(rhs);
-      case RELATIONAL_OPERATOR_STARTS_WITH:
-        return lhs.startsWith(rhs);
-      case RELATIONAL_OPERATOR_ENDS_WITH:
-        return lhs.endsWith(rhs);
-      case RELATIONAL_OPERATOR_REGEX_MATCH:
-        try {
+    try {
+      switch (relationalOperator) {
+        case RELATIONAL_OPERATOR_CONTAINS:
+          return lhs.contains(rhs);
+        case RELATIONAL_OPERATOR_EQUALS:
+          return lhs.equals(rhs);
+        case RELATIONAL_OPERATOR_NOT_EQUALS:
+          return !lhs.equals(rhs);
+        case RELATIONAL_OPERATOR_STARTS_WITH:
+          return lhs.startsWith(rhs);
+        case RELATIONAL_OPERATOR_ENDS_WITH:
+          return lhs.endsWith(rhs);
+        case RELATIONAL_OPERATOR_REGEX_MATCH:
           return Pattern.compile(rhs).matcher(lhs).find();
-        } catch (Exception e) {
-          log.error("Invalid regex: {} passed to match", rhs);
-          if (log.isDebugEnabled()) {
-            log.debug(
-                "Invalid regex passed to match. Hence returning false. lhs: {} and rhs: {}",
-                lhs,
-                rhs);
-          }
+        default:
+          log.error("Unsupported relational operator for string value rhs:{}", relationalOperator);
           return false;
-        }
-      default:
-        log.error("Unsupported relational operator for string value rhs:{}", relationalOperator);
-        return false;
+      }
+    } catch (Exception e) {
+      log.error(
+          "Invalid regex or invalid value for lhs: {} or rhs: {} passed to match. Hence returning false.",
+          lhs,
+          rhs);
+      log.debug(
+          "Invalid regex or invalid value for lhs: {} or rhs: {} passed to match. Hence returning false.",
+          lhs,
+          rhs);
```

Comment:
It's not invalid regex anymore. We should just rephrase it `"Unable to match lhs:{} with rhs:{} for operator: {}"`